src/CommandAndStateMessageParser.cpp

```diff
@@ -252,14 +252,13 @@ string CommandAndStateMessageParser::parseDriveRevolutionCommandMessage(
     string address,
     int model,
     commands::LinearAngular6DCommand command,
-    double minimum_vertical_command)
+    double buoyancy_compensation_offset)
 {
     auto message = payloadSetMessageTemplate(api_version, address, model);
     auto root = message["payload"]["devices"][address];
 
     auto vertical_cmd_sign = command.z() / abs(command.z());
-    auto vertical_cmd =
-        max(abs(command.z()), minimum_vertical_command) * vertical_cmd_sign;
+    auto vertical_cmd = abs(command.z()) * vertical_cmd_sign + buoyancy_compensation_offset;
```

Comment:
This is `abs(command.z()) * command.z() / abs(command.z()) + buoyancy_compensation_offset`. I let you simplify ...